rust/datafusion/src/execution/context.rs

```diff
@@ -1702,6 +1702,23 @@ mod tests {
         }
         Ok(())
     }
+    #[test]
+    fn ctx_sql_should_optimize_plan() -> Result<()> {
+        let mut ctx = ExecutionContext::new();
+        let plan1 =
+            ctx.create_logical_plan("SELECT * FROM (SELECT 1) WHERE TRUE AND TRUE")?;
+
+        let opt_plan1 = ctx.optimize(&plan1)?;
+
+        let plan2 = ctx.sql("SELECT * FROM (SELECT 1) WHERE TRUE AND TRUE")?;
```

Comment:
Before the change the test fails, as it doesn't optimize the plan.

Reply:
👍